pkg/services/ngalert/schedule/schedule.go

```diff
@@ -549,9 +347,21 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key models.AlertRul
 
 		processedStates := sch.stateManager.ProcessEvalResults(ctx, r, results)
 		sch.saveAlertStates(ctx, processedStates)
-		alerts := FromAlertStateToPostableAlerts(processedStates, sch.stateManager, sch.appURL)
-
-		notify(alerts, logger)
+		toNotify := make([]*state.State, 0, len(processedStates))
+		sent := sch.clock.Now()
+		for _, alertState := range toNotify {
+			if !alertState.NeedsSending(sch.stateManager.ResendDelay) {
+				continue
+			}
+			toNotify = append(toNotify, alertState)
+			alertState.LastSentAt = sent
```

Comment:
I know its a `*.state.State` but I think it would avoid confusion if we set `LastSentAt` before `append`?